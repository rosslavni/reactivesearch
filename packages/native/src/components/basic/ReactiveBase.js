import React, { Component } from 'react';
import { Provider } from 'react-redux';
import { Container } from 'native-base';
import Appbase from 'appbase-js';

import configureStore from '@appbaseio/reactivecore';

import types from '@appbaseio/reactivecore/lib/utils/types';

class ReactiveBase extends Component {
	constructor(props) {
		super(props);

		this.type = props.type ? props.type : '*';

		const credentials = props.url && props.url.trim() !== '' && !props.credentials
			? 'user:pass'
			: props.credentials;

		const config = {
			url: props.url && props.url.trim() !== '' ? props.url : 'https://scalr.api.appbase.io',
			app: props.app,
			credentials,
			type: this.type,
		};

		const appbaseRef = new Appbase(config);
		this.store = configureStore({ config, appbaseRef });
	}

	render() {
		return (
			<Provider store={this.store}>
				<Container>
					{this.props.children}
				</Container>
			</Provider>
		);
	}
}

ReactiveBase.propTypes = {
	type: types.string,
	url: types.string,
	credentials: types.string,
	app: types.stringRequired,
	children: types.children,
};

export default ReactiveBase;
