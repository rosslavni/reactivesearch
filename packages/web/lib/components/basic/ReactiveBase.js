var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

import React, { Component } from "react";
import { Provider } from "react-redux";
import Appbase from "appbase-js";

import configureStore from "@appbaseio/reactivecore";
import types from "@appbaseio/reactivecore/lib/utils/types";
import URLParamsProvider from "./URLParamsProvider";

var URLSearchParams = require("url-search-params");

var ReactiveBase = function (_Component) {
	_inherits(ReactiveBase, _Component);

	function ReactiveBase(props) {
		_classCallCheck(this, ReactiveBase);

		var _this = _possibleConstructorReturn(this, _Component.call(this, props));

		_this.type = props.type ? props.type : "*";

		var credentials = props.url && props.url.trim() !== "" && !props.credentials ? "user:pass" : props.credentials;

		var config = {
			url: props.url && props.url.trim() !== "" ? props.url : "https://scalr.api.appbase.io",
			app: props.app,
			credentials: props.credentials,
			type: _this.type
		};

		_this.params = new URLSearchParams(window.location.search);
		var selectedValues = {};

		try {
			for (var _iterator = _this.params.keys(), _isArray = Array.isArray(_iterator), _i = 0, _iterator = _isArray ? _iterator : _iterator[Symbol.iterator]();;) {
				var _extends2;

				var _ref;

				if (_isArray) {
					if (_i >= _iterator.length) break;
					_ref = _iterator[_i++];
				} else {
					_i = _iterator.next();
					if (_i.done) break;
					_ref = _i.value;
				}

				var key = _ref;

				selectedValues = _extends({}, selectedValues, (_extends2 = {}, _extends2[key] = { value: JSON.parse(_this.params.get(key)) }, _extends2));
			}
		} catch (e) {
			selectedValues = {};
		};

		var appbaseRef = new Appbase(config);
		_this.store = configureStore({ config: config, appbaseRef: appbaseRef, selectedValues: selectedValues });
		return _this;
	}

	ReactiveBase.prototype.render = function render() {
		return React.createElement(
			Provider,
			{ store: this.store },
			React.createElement(
				URLParamsProvider,
				{ params: this.params },
				this.props.children
			)
		);
	};

	return ReactiveBase;
}(Component);

ReactiveBase.propTypes = {
	type: types.type,
	url: types.url,
	credentials: types.credentials,
	app: types.app,
	children: types.children
};

export default ReactiveBase;