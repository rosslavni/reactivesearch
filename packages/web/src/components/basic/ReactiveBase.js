import React, { Component } from "react";
import { Provider } from "react-redux";
import Appbase from "appbase-js";

import configureStore from "@appbaseio/reactivecore";
import types from "@appbaseio/reactivecore/lib/utils/types";

import { base } from "../../styles";

class ReactiveBase extends Component {
	constructor(props) {
		super(props);

		this.type = props.type ? props.type : "*";

		const credentials = props.url && props.url.trim() !== "" && !props.credentials
			? "user:pass"
			: props.credentials;

		const config = {
			url: props.url && props.url.trim() !== "" ? props.url : "https://scalr.api.appbase.io",
			app: props.app,
			credentials: props.credentials,
			type: this.type
		};

		const appbaseRef = new Appbase(config);
		this.store = configureStore({ config, appbaseRef });
	}

	render() {
		return (
			<Provider store={this.store}>
				<div className={base}>
					{this.props.children}
				</div>
			</Provider>
		);
	}
}

ReactiveBase.propTypes = {
	type: types.type,
	url: types.url,
	credentials: types.credentials,
	app: types.app,
	children: types.children
}
export default ReactiveBase;
