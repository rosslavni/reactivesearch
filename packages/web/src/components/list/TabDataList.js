import { hasCustomRenderer } from '@appbaseio/reactivecore/lib/utils/helper';
import types from '@appbaseio/reactivecore/lib/utils/types';
import { bool } from 'prop-types';
import React from 'react';
import { TabLink, TabContainer } from '../../styles/Tabs';

import SingleDataList from './SingleDataList';

const TabDataList = (props) => {
	const { renderItem } = props;
	const defaultItem = item =>
		`${item.label} ${props.showCount && item.count ? `(${item.count})` : ''}`;

	if (hasCustomRenderer(props) || props.showRadio) {
		return <SingleDataList {...props} />;
	}
	return (
		<SingleDataList
			{...props}
			showSearch={props.showSearch}
			render={({ data, value, handleChange }) => (
				<TabContainer vertical={props.displayAsVertical}>
					{data.map(item => (
						<TabLink
							onClick={() => handleChange(item.value)}
							selected={item.value === value}
							vertical={props.displayAsVertical}
							key={item.value}
						>
							{renderItem
								? renderItem(item.label, item.count, item.value === value)
								: defaultItem(item)}
						</TabLink>
					))}
				</TabContainer>
			)}
		/>
	);
};

TabDataList.defaultProps = {
	displayAsVertical: false,
	showRadio: false,
	showSearch: false,
};

TabDataList.propTypes = {
	displayAsVertical: bool,
	children: types.func,
	componentId: types.stringRequired,
	dataField: types.stringRequired,
	onChange: types.func,
	react: types.react,
	title: types.title,
	URLParams: types.bool,
	showCount: types.bool,
	showRadio: types.bool,
	showSearch: types.bool,
	render: types.func,
	renderItem: types.func,
	renderNoResults: types.func,
	index: types.string,
};
export default TabDataList;
