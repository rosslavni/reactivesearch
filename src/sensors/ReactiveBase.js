import React, { Component } from "react";
import { Provider } from "react-redux";
import { View } from "react-native";

import configureStore from "../store/configureStore";

export default class ReactiveBase extends Component {
	constructor(props) {
		super(props);

		this.type = props.type ? props.type : "*";
		const config = {
			url: "scalr.api.appbase.io",
			app: props.app,
			credentials: props.credentials,
			type: this.type
		};
		this.store = configureStore({config});
	}

	render() {
		return (
			<Provider store={this.store}>
				<View style={this.props.style}>
					{this.props.children}
				</View>
			</Provider>
		);
	}
}
