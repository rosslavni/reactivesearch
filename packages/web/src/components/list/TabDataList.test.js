import React from 'react';
import renderer from 'react-test-renderer';
import ReactiveBase from '../basic/ReactiveBase';
import TabDataList from './TabDataList';

const MOCK_AGGREGATIONS_DATA = {
	'authors.keyword': {
		buckets: [
			{
				key: 'J. K. Rowling',
				doc_count: 10,
			},
			{
				key: 'Nora Roberts',
				doc_count: 7,
			},
			{
				key: 'Bram Stoker',
				doc_count: 12,
			},
		],
	},
};
it('should render', () => {
	const elem = renderer
		.create(
			<ReactiveBase app="test" url="https://foo:bar@localhost:800">
				<TabDataList
					mode="test"
					componentId="authors"
					dataField="authors.keyword"
					data={[
						{ label: 'J. K. Rowling', value: 'J. K. Rowling' },
						{ label: 'Nora Roberts', value: 'Nora Roberts' },
					]}
					mockData={{
						aggregations: MOCK_AGGREGATIONS_DATA,
					}}
				/>
			</ReactiveBase>,
		)
		.toJSON();
	expect(elem).toMatchSnapshot();
});
it('should render with title', () => {
	const elem = renderer
		.create(
			<ReactiveBase app="test" url="https://foo:bar@localhost:800">
				<TabDataList
					mode="test"
					componentId="authors"
					dataField="authors.keyword"
					data={[
						{ label: 'J. K. Rowling', value: 'J. K. Rowling' },
						{ label: 'Nora Roberts', value: 'Nora Roberts' },
					]}
					mockData={{
						aggregations: MOCK_AGGREGATIONS_DATA,
					}}
				/>
			</ReactiveBase>,
		)
		.toJSON();
	expect(elem).toMatchSnapshot();
});

it('should render with renderItem', () => {
	const elem = renderer
		.create(
			<ReactiveBase app="test" url="https://foo:bar@localhost:800">
				<TabDataList
					mode="test"
					componentId="authors"
					dataField="authors.keyword"
					data={[
						{ label: 'J. K. Rowling', value: 'J. K. Rowling' },
						{ label: 'Nora Roberts', value: 'Nora Roberts' },
					]}
					mockData={{
						aggregations: MOCK_AGGREGATIONS_DATA,
					}}
					renderItem={item => `${item.label} V`}
				/>
			</ReactiveBase>,
		)
		.toJSON();
	expect(elem).toMatchSnapshot();
});

it('should render with showCount', () => {
	const elem = renderer
		.create(
			<ReactiveBase app="test" url="https://foo:bar@localhost:800">
				<TabDataList
					mode="test"
					componentId="authors"
					dataField="authors.keyword"
					data={[
						{ label: 'J. K. Rowling', value: 'J. K. Rowling' },
						{ label: 'Nora Roberts', value: 'Nora Roberts' },
					]}
					mockData={{
						aggregations: MOCK_AGGREGATIONS_DATA,
					}}
					showCount
				/>
			</ReactiveBase>,
		)
		.toJSON();
	expect(elem).toMatchSnapshot();
});
