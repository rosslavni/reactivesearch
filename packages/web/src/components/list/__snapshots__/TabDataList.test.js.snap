// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
.emotion-0 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI","Roboto","Noto Sans","Ubuntu","Droid Sans","Helvetica Neue",sans-serif;
  font-size: 16px;
  color: #424242;
  width: 100%;
}

.emotion-0 input,
.emotion-0 button,
.emotion-0 textarea,
.emotion-0 select {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI","Roboto","Noto Sans","Ubuntu","Droid Sans","Helvetica Neue",sans-serif;
}

.emotion-0 *,
.emotion-0 *:before,
.emotion-0 *:after {
  box-sizing: border-box;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  max-width: 90vw;
  overflow-x: auto;
}

.emotion-6 {
  color: #424242;
  font-weight: normal;
  padding: 0.5rem 0.5rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-color: white;
  border-width: 0px;
  border-style: solid;
  border-bottom-width: 0px;
  border-left-width: 0px;
  display: inline-block;
  cursor: pointer;
  white-space: nowrap;
}

<div
  className="emotion-0 emotion-1"
  style={Object {}}
>
  <div
    className="emotion-2 emotion-3"
    style={Object {}}
  >
    <div
      className="emotion-4 emotion-5"
    >
      <a
        className="emotion-6 emotion-7"
        onClick={[Function]}
        selected={false}
      >
        J. K. Rowling 
      </a>
      <a
        className="emotion-6 emotion-7"
        onClick={[Function]}
        selected={false}
      >
        Nora Roberts 
      </a>
    </div>
  </div>
</div>
`;

exports[`should render with renderItem 1`] = `
.emotion-0 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI","Roboto","Noto Sans","Ubuntu","Droid Sans","Helvetica Neue",sans-serif;
  font-size: 16px;
  color: #424242;
  width: 100%;
}

.emotion-0 input,
.emotion-0 button,
.emotion-0 textarea,
.emotion-0 select {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI","Roboto","Noto Sans","Ubuntu","Droid Sans","Helvetica Neue",sans-serif;
}

.emotion-0 *,
.emotion-0 *:before,
.emotion-0 *:after {
  box-sizing: border-box;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  max-width: 90vw;
  overflow-x: auto;
}

.emotion-6 {
  color: #424242;
  font-weight: normal;
  padding: 0.5rem 0.5rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-color: white;
  border-width: 0px;
  border-style: solid;
  border-bottom-width: 0px;
  border-left-width: 0px;
  display: inline-block;
  cursor: pointer;
  white-space: nowrap;
}

<div
  className="emotion-0 emotion-1"
  style={Object {}}
>
  <div
    className="emotion-2 emotion-3"
    style={Object {}}
  >
    <div
      className="emotion-4 emotion-5"
    >
      <a
        className="emotion-6 emotion-7"
        onClick={[Function]}
        selected={false}
      >
        J. K. Rowling V
      </a>
      <a
        className="emotion-6 emotion-7"
        onClick={[Function]}
        selected={false}
      >
        Nora Roberts V
      </a>
    </div>
  </div>
</div>
`;

exports[`should render with showCount 1`] = `
.emotion-0 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI","Roboto","Noto Sans","Ubuntu","Droid Sans","Helvetica Neue",sans-serif;
  font-size: 16px;
  color: #424242;
  width: 100%;
}

.emotion-0 input,
.emotion-0 button,
.emotion-0 textarea,
.emotion-0 select {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI","Roboto","Noto Sans","Ubuntu","Droid Sans","Helvetica Neue",sans-serif;
}

.emotion-0 *,
.emotion-0 *:before,
.emotion-0 *:after {
  box-sizing: border-box;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  max-width: 90vw;
  overflow-x: auto;
}

.emotion-6 {
  color: #424242;
  font-weight: normal;
  padding: 0.5rem 0.5rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-color: white;
  border-width: 0px;
  border-style: solid;
  border-bottom-width: 0px;
  border-left-width: 0px;
  display: inline-block;
  cursor: pointer;
  white-space: nowrap;
}

<div
  className="emotion-0 emotion-1"
  style={Object {}}
>
  <div
    className="emotion-2 emotion-3"
    style={Object {}}
  >
    <div
      className="emotion-4 emotion-5"
    >
      <a
        className="emotion-6 emotion-7"
        onClick={[Function]}
        selected={false}
      >
        J. K. Rowling (10)
      </a>
      <a
        className="emotion-6 emotion-7"
        onClick={[Function]}
        selected={false}
      >
        Nora Roberts (7)
      </a>
    </div>
  </div>
</div>
`;

exports[`should render with title 1`] = `
.emotion-0 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI","Roboto","Noto Sans","Ubuntu","Droid Sans","Helvetica Neue",sans-serif;
  font-size: 16px;
  color: #424242;
  width: 100%;
}

.emotion-0 input,
.emotion-0 button,
.emotion-0 textarea,
.emotion-0 select {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI","Roboto","Noto Sans","Ubuntu","Droid Sans","Helvetica Neue",sans-serif;
}

.emotion-0 *,
.emotion-0 *:before,
.emotion-0 *:after {
  box-sizing: border-box;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  max-width: 90vw;
  overflow-x: auto;
}

.emotion-6 {
  color: #424242;
  font-weight: normal;
  padding: 0.5rem 0.5rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-color: white;
  border-width: 0px;
  border-style: solid;
  border-bottom-width: 0px;
  border-left-width: 0px;
  display: inline-block;
  cursor: pointer;
  white-space: nowrap;
}

<div
  className="emotion-0 emotion-1"
  style={Object {}}
>
  <div
    className="emotion-2 emotion-3"
    style={Object {}}
  >
    <div
      className="emotion-4 emotion-5"
    >
      <a
        className="emotion-6 emotion-7"
        onClick={[Function]}
        selected={false}
      >
        J. K. Rowling 
      </a>
      <a
        className="emotion-6 emotion-7"
        onClick={[Function]}
        selected={false}
      >
        Nora Roberts 
      </a>
    </div>
  </div>
</div>
`;
